### PR TITLE
Laboratory work #1, Egor Solovyev - 22FPL2

### PR DESCRIPTION
**Pull request by egor-solo**

### CODE REVIEW

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]
Good start, hope you will be in time

File: lab_1_classify_by_unigrams/main.py [hunk 1]
```diff
@@ -11,6 +11,16 @@ def tokenize(text: str) -> list[str] | None:
     :param text: a text
     :return: a list of lower-cased tokens without punctuation
     """
+    if not isinstance(text, str):
+        return None
+    else:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:16]: you do not `else` if you return in if

File: lab_1_classify_by_unigrams/main.py [hunk 2]
```diff
@@ -11,6 +11,16 @@ def tokenize(text: str) -> list[str] | None:
...
+        tokens = (text.lower())
+        punctuation = " !#$%&'\'()*+,-./:;<=>?@[\\]^_`{|}~1234567890"
+        for token in tokens:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:19]: try to do it in one line

File: lab_1_classify_by_unigrams/main.py [hunk 3]
```diff
@@ -19,6 +29,17 @@ def calculate_frequencies(tokens: list[str] | None) -> dict[str, float] | None:
     :param tokens: a list of tokens
     :return: a dictionary with frequencies
     """
+    freq = {}
+    if not isinstance(tokens, list):
+        return None
+    for token in tokens:
```
Comment (artyomtugaryov) [inline lab_1_classify_by_unigrams/main.py:35]: too many fors - merge it

File: main.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
+
```
Comment (artyomtugaryov) [inline main.py:1]: what is it? remove!

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: main.py (hunk 4 above)
Comment (egor-solo) [reply main.py]: removed

File: lab_1_classify_by_unigrams/main.py (hunk 3 above)
Comment (egor-solo) [reply lab_1_classify_by_unigrams/main.py:30]: fixed

File: lab_1_classify_by_unigrams/main.py (hunk 2 above)
Comment (egor-solo) [reply lab_1_classify_by_unigrams/main.py]: done

File: lab_1_classify_by_unigrams/main.py (hunk 1 above)
Comment (egor-solo) [reply lab_1_classify_by_unigrams/main.py]: removed

**Pull request closed without merging**